### PR TITLE
Enhanced support for selection ranges

### PR DESCRIPTION
**Pull request by miguel-guedes**
* HH should now fully support the available two types of selection `Range`s (see second paragraph [here](https://developer.mozilla.org/en-US/docs/Web/API/Range/startOffset) for context)
* Tweaked & fixed a few aspects about the build
* Updated README
* Added master index file so anyone jumping in after `yarn run start`ing the thing is taken to the existing working example

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -110,34 +110,6 @@ const assets = {
   },
 };
 
-const tests = {
```
Comment (kujenga) [inline webpack.config.js]: Are these remove tests something we should be looking to replace? Or are they just being run already in a different manner?

Comment (miguel-guedes) [reply webpack.config.js]: @kujenga The tests can be run in the terminal. (they're part of CI currently)

**Comment by miguel-guedes** [comment]
I can add tests, if you like, @kujenga. Let me know!

**Comment by kujenga** [comment]
@miguel-guedes yes, let’s! I believe I’ve also resolved the issue with the npm failure in the build.

**Comment by miguel-guedes** [comment]
It is not possible to craft tests for the changes in this PR because jsdom does not support the `createRange` method that is part of the standard Web APIs.  References:

https://github.com/dossier/html-highlighter/pull/33#issuecomment-358652335
https://github.com/jsdom/jsdom/issues/317

@kujenga

**Comment by kujenga** [comment]
Ah, right! Bummer. This can come in as-is then.

@miguel-guedes would we be able to leverage puppeteer and headless chrome here maybe to cover these cases that jsdom currently does not? For future work.

**Comment by miguel-guedes** [comment]
> @miguel-guedes would we be able to leverage puppeteer and headless chrome here maybe to cover these cases that jsdom currently does not? For future work.

Most definitely, yes.

**Pull request merged**